### PR TITLE
[DH-541] Support Elasticsearch 5.5.

### PR DESCRIPTION
**Pull request by elcct**
Biggest difference is that Elasticsearch 5 has `fielddata` set to `False` by default. It means that we have to set it to `True` for the string fields we want to sort by.

### CODE REVIEW

File: datahub/search/dsl_utils.py [hunk 1]
```diff
@@ -1,11 +1,16 @@
 from functools import partial
-from elasticsearch_dsl import Nested, String
+from elasticsearch_dsl import Keyword, Nested, String
 
 
-KeywordString = partial(String, index='not_analyzed')
-CaseInsensitiveKeywordString = partial(String, analyzer='lowercase_keyword_analyzer')
+KeywordString = Keyword
```
Comment (reupen) [inline datahub/search/dsl_utils.py]: Do we need to keep this?

File: datahub/search/dsl_utils.py [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+CaseInsensitiveKeywordString = partial(
```
Comment (reupen) [inline datahub/search/dsl_utils.py]: Should 'Sortable' be added to the name here?

**Comment by reupen** [comment]
Do we have ES 5.5 instances on standby ready to switch to? Should we change the name of the ES_URL environment variable to e.g. ES5_URL, just thinking it might make deployment easier.

**Comment by marcofucci** [comment]
I don't know exactly why you started to use `fielddata` but the docs say that we should really avoid them as 
> fielddata can consume a lot of heap space

As I understand, fielddata is just for rare cases. Most of the times, if you need sorting or aggregations on text, it's better to use `keywords` which support `doc_values`.

Why don't we use `raw` https://www.elastic.co/guide/en/elasticsearch/guide/master/multi-fields.html for sorting?

File: datahub/search/dsl_utils.py (hunk 1 above)
Comment (elcct) [reply datahub/search/dsl_utils.py]: We don't. I will remove it.

File: datahub/search/dsl_utils.py (hunk 2 above)
Comment (elcct) [reply datahub/search/dsl_utils.py]: I had the same question, so I will add it. Thanks

**Comment by elcct** [comment]
@marcofucci in 2.4 `fielddata` is enabled by default on those fields. I think the first step would be to get compatibility first and then do optimisations specific to the platform.
Regarding multi fields we just use multiple fields in the model as this was more convenient (in a `fieldname_variation` format. As a next step we could add a keyword field for the purpose of sorting. 
But if we want to have case insensitive sorting, then we need to keep `fielddata`, as `Keyword` is case sensitive.

**Comment by elcct** [comment]
@reupen I don't know about instance. But having new env variable sounds good. I will add that.

**Comment by marcofucci** [comment]
@elcct 👍  makes sense to refactor in steps. 
In 5.x there seem to be a new [normalizer](https://discuss.elastic.co/t/best-practice-of-case-insensitive-keyword-mapping-in-es-5-x/74015) thing which we could use.

**Pull request merged**